src/client/opamCommands.ml

```diff
@@ -2894,6 +2907,10 @@ let switch cli =
            order = `Custom order;
         }
         compilers;
+      if not all then
```

Comment:
It is better to show the message only if there was some hidden packages.
Ftm, we don't have the information. If we retrieve the complete `compilers` set and filter it afterwards compilers flagged with `avoid version` or `deprecated`, we can access to the diff.

Reply:
done